controllers/ocicluster_controller.go

```diff
@@ -106,9 +100,41 @@ func (r *OCIClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	var clusterScope scope.ClusterScopeClient
+	regionOverride := r.Region
 
-	clients, err := r.ClientProvider.GetOrBuildClient(regionOverride)
+	var clusterScope scope.ClusterScopeClient
+	var clients scope.OCIClients
+	var clientProvider *scope.ClientProvider
+	if ociCluster.Spec.IdentityRef != nil {
+		identity, err := cloudutil.GetClusterIdentityFromRef(ctx, r.Client, ociCluster.Namespace, ociCluster.Spec.IdentityRef)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if !cloudutil.IsClusterNamespaceAllowed(ctx, r.Client, identity.Spec.AllowedNamespaces, ociCluster.Namespace) {
+			conditions.MarkFalse(ociCluster, infrastructurev1beta1.ClusterReadyCondition, infrastructurev1beta1.NamespaceNotAllowedByIdentity, clusterv1.ConditionSeverityError, "")
```

Comment:
thoughts on adding which namespace is causing the issue?
Maybe add the allow list and the namespace to the condition message 🤷 

I'm only bringing this up because I think as a project we may want to do more of this kind of thing with the conditions.

Reply:
fixed